spotbugs/src/main/java/edu/umd/cs/findbugs/SAXBugCollectionHandler.java

```diff
@@ -573,7 +578,38 @@ private void parseBugInstanceContents(String qName, Attributes attributes) throw
         } else if ("UserAnnotation".equals(qName)) {
             // legacy from the cloud plugin stuff
         } else {
-            throw new SAXException("Unknown bug annotation named " + qName);
+            // @Anemone, add custom bug annotation types,
+            // which can be deserialized with 'fromXML' method in its class.
+            Method fromXML = qnameCache.computeIfAbsent(qName, k -> {
+                for (Plugin plugin : Plugin.getAllPlugins()) {
+                    Class<?> annotationClazz;
+                    try {
+                        // The qName should equal to its classname, so there can reflect.
+                        annotationClazz = plugin.getClassLoader().loadClass(k);
+                        return annotationClazz.getMethod("fromXML", String.class, Attributes.class);
+                    } catch (NoSuchMethodException | ClassCastException e) {
+                        e.printStackTrace();
+                    } catch (ClassNotFoundException ignored) {
+                        LOG.warn(k + " not found in Plugin(" + plugin.getPluginId() + ")");
```

Comment:
This should use parameterised logging instead of concatenating strings.

Reply:
Done